src/codegen/pass2_generate.cc

```diff
@@ -1451,6 +1451,19 @@ static void gen_cond_enum(Scratchbuf& buf,
                 }
                 append(block, code_text(alc, buf.flush()));
             }
+        } else if (opts->lang == Lang::D) {
+            start = buf.cstr("enum ").str(opts->api_cond_type).cstr(" {").flush();
+            end = "};";
+            for (const StartCond& cond : conds) {
+                buf.str(cond.name);
+                if (opts->loop_switch) {
+                    buf.cstr(" = ").u32(cond.number);
+                }
+                if (&cond != last_cond) { // do not append comma after the last one
+                    buf.cstr(",");
+                }
+                append(block, code_text(alc, buf.flush()));
+            }
```

Comment:
Absolutely, I had  duplicated a bunch of the code in case I discovered I needed to tweak it, but things have been working perfectly for the examples I have tried so I can eliminate those.